### PR TITLE
tests: switch unittest to pytest

### PR DESCRIPTION
**Pull request by kkrolczyk**
two tests are commented out, it seems
that there is no easy way to test if fixture
raises intended error. I've tested a couple
of not-so-easy-hackery-ways but to no avail.

Signed-off-by: Krzysztof Królczyk <Krzysztof.Krolczyk@o2.pl>

### CODE REVIEW

**Comment by priv-kweihmann** [comment]
Could you please add `Closes #210 ` or `Relates to #210` into the commit message, so we could track the issue better

File: tests/test_license.py [hunk 1]
```diff
@@ -1,58 +1,75 @@
-#!/usr/bin/env python3
-
-# SPDX-FileCopyrightText: 2022 Henrik Sandklef
+# SPDX-FileCopyrightText: 2022 we, the people
```
Comment (priv-kweihmann) [inline tests/test_license.py]: Nope - either add your own CR as an additional line or keep it as it is

File: tests/test_license.py [hunk 2]
```diff
@@ -1,58 +1,75 @@
...
+    assert all((lic["type"] == "license" for lic in p["operands"]))
+    assert all((lic["name"] in ("MIT", "X11") for lic in p["operands"]))
 
-        operands = mit_and_x11['license']['operands']
 
-        self.assertTrue(operands[0]['name'] == "MIT" and operands[1]['name'] == "X11" or
-                            operands[1]['name'] == "MIT" and operands[0]['name'] == "X11")
+@pytest.mark.parametrize('lic_expr', ["This is not a list", ])
+def test_fail_on_invalid_str_license_expression(parse_simple):
+    # with pytest.raises(FlictError):  #  FIXME flict does not fail here
```
Comment (priv-kweihmann) [inline tests/test_license.py]: Huch, really? That's worth a new bug ticket

**Review by priv-kweihmann — COMMENTED** [review]
LGTM expect for the CR thingy - the commented out test can be fixed as part of a new bug ticket, as I believe it's an actual bug in the implementation

File: tests/test_license.py (hunk 1 above)
Comment (jens-erdmann) [reply tests/test_license.py]: But it was funny, though.

Comment (kkrolczyk) [reply tests/test_license.py]: that was the idea ;)

**Comment by kkrolczyk** [comment]
```python                                                                                                                                                                                                        
@pytest.mark.parametrize('lic_expr', ["This is not a list", ])
def test_fail_on_invalid_str_license_expression(parse_simple):
    with pytest.raises(FlictError):  #  FIXME flict does not fail here
        pass

@pytest.mark.parametrize('lic_expr', [[], ])
def test_fail_on_empty_license_expression(parse_simple):
    with pytest.raises(FlictError) as err:
        pass
    assert err.value.code == ReturnCodes.RET_INVALID_EXPRESSSION


@pytest.mark.parametrize('lic_expr', [[""], ])
def test_fail_on_empty_str_license_expression(parse_simple):
    with pytest.raises(TypeError):
        pass

```

Removed those - will create an issue to look into it.

File: tests/test_license.py (hunk 2 above)
Comment (kkrolczyk) [reply tests/test_license.py]: yup, it surprised me to that degeree that i didn't trust pytest and ran
```shell
$ flict simplify "This is not a list"
{"original": "This is not a list", "simplified": "This is not a list"}
```

Comment (priv-kweihmann) [reply tests/test_license.py]: @hesa your turn

Comment (jens-erdmann) [reply tests/test_license.py]: I guess #231 fulfills this request.

**Review by jens-erdmann — APPROVED** [review]
Good job.

File: tests/test_license.py (hunk 2 above)
Comment (priv-kweihmann) [reply tests/test_license.py]: Indeed it does

**Pull request merged**